DNN Platform/Library/Services/Installer/Installers/UrlProviderInstaller.cs

```diff
@@ -60,13 +60,8 @@ public override string AllowableFiles
         private void DeleteProvider()
         {
             try
-            {
-                //Attempt to get the Desktop Module
-                //Attempt to get the Desktop Module
-                DesktopModuleInfo desktopModule = DesktopModuleController.GetDesktopModuleByPackageID(Package.PackageID);
-
-                ExtensionUrlProviderInfo tempUrlProvider = ExtensionUrlProviderController.GetProviders(Null.NullInteger)
-                                            .SingleOrDefault(p => p.DesktopModuleId == desktopModule.DesktopModuleID);
+            {
+				ExtensionUrlProviderInfo tempUrlProvider = ExtensionUrlProviderController.GetProviders(Null.NullInteger).Where(p => p.ProviderName == _extensionUrlProvider.ProviderName && p.ProviderType == _extensionUrlProvider.ProviderType).FirstOrDefault();
```

Comment:
Unless there's a reason otherwise, I think I'd prefer to keep this using `SingleOrDefault`:

    ExtensionUrlProviderInfo tempUrlProvider = ExtensionUrlProviderController.GetProviders(Null.NullInteger)
        .SingleOrDefault(p => p.ProviderName == _extensionUrlProvider.ProviderName && p.ProviderType == _extensionUrlProvider.ProviderType);

Otherwise looks good.  Do you know why `Package.PackageID` was throwing a null reference?  Is this running after the package is uninstalled?